### PR TITLE
Pass AdoptDefaults and buildConfig CI/BUILD_REF overrides to RepoHandler

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Pass knowledge of build REF overrides and AdoptDefaults json to the RepoHandler
And ensure all set correctly from ci-jenkins-pipelines

Fixes: https://github.com/adoptium/ci-jenkins-pipelines/issues/603

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
run tests

**Review by karianna — APPROVED** [review]
LGTM

**Review by karianna — APPROVED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1466,13 +1484,16 @@ class Build {
                                     repoHandler.setUserDefaultsJson(context, DEFAULTS_JSON)
                                     repoHandler.checkoutUserPipelines(context)
                                 }
+                                printGitRepoInfo()
                             } else {
                                 context.println "[CHECKOUT] Checking out to the user's temurin-build..."
                                 repoHandler.setUserDefaultsJson(context, DEFAULTS_JSON)
                                 repoHandler.checkoutUserBuild(context)
+                                printGitRepoInfo()
```
Comment (zdtsw) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1492]: do we want all these call to printGitRepoInfo() or most of them are for debug purpose?

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1492]: So yes, I think it is useful as it shows clearly what you've built with, and especially if there are "Untracked" files left around. It's only a few lines. The trouble we've had in the past in not clearly knowing for sure we've built with the right scripts...

**Review by karianna — APPROVED** [review]

**Review by zdtsw — APPROVED** [review]

**Pull request merged**